contracts/shapella-upgrade/ShapellaUpgradeTemplate.sol

```diff
@@ -0,0 +1,870 @@
+    /// @dev Represented in the Basis Points (100% == 10_000)
+    uint256 oneOffCLBalanceDecreaseBPLimit;
+
+    /// @notice The max annual increase of the total validators' balances on the Consensus Layer
+    ///     since the previous oracle report
+    /// @dev Represented in the Basis Points (100% == 10_000)
+    uint256 annualBalanceIncreaseBPLimit;
+
+    /// @notice The max deviation of the provided `simulatedShareRate`
+    ///     and the actual one within the currently processing oracle report
+    /// @dev Represented in the Basis Points (100% == 10_000)
+    uint256 simulatedShareRateDeviationBPLimit;
+
+    /// @notice The max number of exit requests allowed in report to ValidatorsExitBusOracle
+    uint256 maxValidatorExitRequestsPerReport;
+
+    /// @notice The max number of data list items reported to accounting oracle in extra data
+    /// @dev Must fit into uint16 (<= 65_535)
+    uint256 maxAccountingExtraDataListItemsCount;
+
+    /// @notice The max number of node operators reported per extra data list item
+    /// @dev Must fit into uint16 (<= 65_535)
+    uint256 maxNodeOperatorsPerExtraDataItemCount;
+
+    /// @notice The min time required to be passed from the creation of the request to be
+    ///     finalized till the time of the oracle report
+    uint256 requestTimestampMargin;
+
+    /// @notice The positive token rebase allowed per single LidoOracle report
+    /// @dev uses 1e9 precision, e.g.: 1e6 - 0.1%; 1e9 - 100%, see `setMaxPositiveTokenRebase()`
+    uint256 maxPositiveTokenRebase;
+}
+
+enum StakingModuleStatus {
+    Active, // deposits and rewards allowed
+    DepositsPaused, // deposits NOT allowed, rewards allowed
+    Stopped // deposits and rewards NOT allowed
+}
+
+struct StakingModule {
+    /// @notice unique id of the staking module
+    uint24 id;
+    /// @notice address of staking module
+    address stakingModuleAddress;
+    /// @notice part of the fee taken from staking rewards that goes to the staking module
+    uint16 stakingModuleFee;
+    /// @notice part of the fee taken from staking rewards that goes to the treasury
+    uint16 treasuryFee;
+    /// @notice target percent of total validators in protocol, in BP
+    uint16 targetShare;
+    /// @notice staking module status if staking module can not accept the deposits or can participate in further reward distribution
+    uint8 status;
+    /// @notice name of staking module
+    string name;
+    /// @notice block.timestamp of the last deposit of the staking module
+    /// @dev NB: lastDepositAt gets updated even if the deposit value was 0 and no actual deposit happened
+    uint64 lastDepositAt;
+    /// @notice block.number of the last deposit of the staking module
+    /// @dev NB: lastDepositBlock gets updated even if the deposit value was 0 and no actual deposit happened
+    uint256 lastDepositBlock;
+    /// @notice number of exited validators
+    uint256 exitedValidatorsCount;
+}
+
+/**
```

Comment:
Thank you for providing the comments section. It's helpful, indeed.

I would suggest extending it vastly for the mainnet upgrade.